Library/Homebrew/formula.rb

```diff
@@ -1625,6 +1625,49 @@ def extract_macho_slice_from(file, arch = Hardware::CPU.arch)
   end
   private :extract_macho_slice_from
 
+  # Generate shell completions for a formula for bash, zsh, and fish, using the formula's binary.
+  #
+  # @param base_name [String] the base name of the generated completion script. Defaults to the formula name.
+  # @param shells [Array<Symbol>] the shells to generate completion scripts for. Defaults to `[:bash, :zsh, :fish]`.
+  # @param binary [Pathname] the binary to use for generating the completion scripts. Defaults to the binary with the
+  # name of the formula.
+  # @param cmd [String] the command to pass to the `binary`. Defaults to 'completion'.
+  # @param shell_as_flag [Boolean] specify if `shells` should each be passed as flags to the `binary`.
+  # Defaults to `false`.
+  sig {
+    params(base_name: String, shells: T::Array[Symbol], binary: Pathname, cmd: String,
+           shell_prefix: T.nilable(T.any(Symbol, String))).void
+  }
+  def generate_completions_from_executable(base_name: name,
+                                           shells: [:bash, :zsh, :fish],
+                                           executable: bin/base_name,
+                                           cmd: "completion",
+                                           shell_parameter_format: nil)
```

Comment:
Thoughts on `shell_parameter_prefix` as a name? That way the string case discussed above is clearer. Also, it's clearer what the difference between `nil` and `:none` is

Reply:
`_prefix` was the name I originally went with, but it does not quite fit, especially with `:none`, as that will not pass anything to the command (as opposed to the default, `nil`, that will simply pass the shell name). 
The `:none` case is needed for the formulae that infer the shell to generate for from the environment variable SHELL.

But I am open to other naming suggestions for this argument!

Reply:
Yeah, fair point. I guess I'm okay with this or the above suggestions since I can't think of something better right now